test/e2e/self_hosted.go

```diff
@@ -226,6 +226,21 @@ func SelfHostedSpec(ctx context.Context, inputGetter func() SelfHostedSpecInput)
 			Namespace:            namespace.Name,
 		})
 
+		log.Logf("Ensure clusterctl does not take ownership on any fields on the self-hosted cluster")
+		selfHostedCluster := framework.GetClusterByName(ctx, framework.GetClusterByNameInput{
+			Getter:    selfHostedClusterProxy.GetClient(),
+			Name:      cluster.Name,
+			Namespace: selfHostedNamespace.Name,
+		})
+		hasClusterctlManagedFields := false
+		for _, managedField := range selfHostedCluster.GetManagedFields() {
+			if managedField.Manager == "clusterctl" {
+				hasClusterctlManagedFields = true
+				break
+			}
+		}
+		Expect(hasClusterctlManagedFields).To(BeFalse())
```

Comment:
maybe add an explanation with the optionalDescription vararg

Otherwise lgtm

Reply:
Hm. Wondering if this will break with older released clusterctl versions (I guess it should as they don't have the fix?)

Let's wait for the test results

Reply:
Ah my bad the self-hosted test only used the latest code

Mixed it up with the clusterctl upgrade test where we use multiple clusterctl versions